File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    hash_map = {}
    for num in phone_book:
        hash_map[num] = 1
    for num in phone_book:
        temp = ""
        for n in num:
            temp+=n
            if temp in hash_map and temp != num:
                return False
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    unordered_set <string> number;
    
    for(auto s:phone_book){
        number.insert(s);
    }
    for(auto num: number){
        string prefix = "";
        for (int i=0; i<num.size()-1;i++){
            prefix+=num[i];
            if(number.contains(prefix)){
                return false;
            }
        }
    }
    return true;
}